plugins/inputs/dpdk/dpdk_test.go

```diff
@@ -0,0 +1,330 @@
+
+		commands := dpdk.getCommandsAndParamsCombinations(ethdevCommands, []string{}, ethdevListCommand, mockAcc)
+
+		require.Equal(t, 4, len(commands))
+		require.Equal(t, 0, len(mockAcc.Errors))
+		require.ElementsMatch(t, commands, expectedCommands)
+	})
+
+	t.Run("when 2 ethdev commands are enabled but one command is disabled, then numberOfIds new commands should be appended", func(t *testing.T) {
+		mockConn, dpdk, mockAcc := prepareEnvironment()
+		defer mockConn.AssertExpectations(t)
+		response := fmt.Sprintf(`{"%s": [1, 123]}`, ethdevListCommand)
+		simulateResponse(mockConn, response, nil)
+		ethdevCommands := []string{"/ethdev/stats", "/ethdev/xstats"}
+		excludedCommands := []string{"/ethdev/xstats"}
+		expectedCommands := []string{"/ethdev/stats,1", "/ethdev/stats,123"}
+
+		commands := dpdk.getCommandsAndParamsCombinations(ethdevCommands, excludedCommands, ethdevListCommand, mockAcc)
+
+		require.Equal(t, 2, len(commands))
+		require.Equal(t, 0, len(mockAcc.Errors))
+		require.ElementsMatch(t, commands, expectedCommands)
+	})
+
+	t.Run("when ethdev commands are enabled but params fetching command returns error then error should be logged in accumulator", func(t *testing.T) {
+		mockConn, dpdk, mockAcc := prepareEnvironment()
+		defer mockConn.AssertExpectations(t)
+		simulateResponse(mockConn, `{notAJson}`, fmt.Errorf("some error"))
+		ethdevCommands := []string{"/ethdev/stats", "/ethdev/xstats"}
+
+		commands := dpdk.getCommandsAndParamsCombinations(ethdevCommands, []string{}, ethdevListCommand, mockAcc)
+
+		require.Equal(t, 0, len(commands))
+		require.Equal(t, 1, len(mockAcc.Errors))
+	})
+}
+
+func Test_Gather(t *testing.T) {
+	t.Run("When parsing a plain json without nested object, then it key should equal to \"\"", func(t *testing.T) {
+		mockConn, dpdk, mockAcc := prepareEnvironment()
+		defer mockConn.AssertExpectations(t)
+		dpdk.AdditionalCommands = []string{"/endpoint1"}
+		simulateResponse(mockConn, `{"/endpoint1":"myvalue"}`, nil)
+
+		err := dpdk.Gather(mockAcc)
+
+		require.NoError(t, err)
+		require.Equal(t, 0, len(mockAcc.Errors))
+		require.Equal(t, 1, len(mockAcc.Metrics))
+		require.Equal(t, "myvalue", mockAcc.Metrics[0].Fields[""])
+	})
+
+	t.Run("When parsing a list of value in nested object then list should be flattened", func(t *testing.T) {
+		mockConn, dpdk, mockAcc := prepareEnvironment()
+		defer mockConn.AssertExpectations(t)
+		dpdk.AdditionalCommands = []string{"/endpoint1"}
+		simulateResponse(mockConn, `{"/endpoint1":{"myvalue":[0,1,123]}}`, nil)
+
+		err := dpdk.Gather(mockAcc)
+
+		require.NoError(t, err)
+		require.Equal(t, 0, len(mockAcc.Errors))
+		require.Equal(t, 1, len(mockAcc.Metrics))
+		require.Equal(t, float64(0), mockAcc.Metrics[0].Fields["myvalue_0"])
+		require.Equal(t, float64(1), mockAcc.Metrics[0].Fields["myvalue_1"])
+		require.Equal(t, float64(123), mockAcc.Metrics[0].Fields["myvalue_2"])
```

Comment:
Please test the whole metric including name and tags. You can do this by defining a reference metric (e.g. call it `expected`) and the compare against your result using `testutil.RequireMetricEqual()` or `testutil.RequireMetricsEqual()`. You probably need to specify the option `IgnoreTime()` as you don't have a deterministic gathering time I think.

Reply:
Nice, rewritten.